layers/cmd/ingest_csv.go

```diff
@@ -101,62 +101,88 @@ func (ci *CSVIngester) Run(pipeline *pipeline.PipelineContext) error {
 		}
 		reader.Comma = rune(ci.Delimiter[0])
 
-		for row := 0; ; row++ {
-			rowData, err := reader.Read()
-			if err == io.EOF {
+		// panic recovery here
+		func() {
```

Comment:
No. Deferred func should only do panic/recover, and inside the for loop. See the pattern in the ticket.